target/linux/generic/files/drivers/platform/mikrotik/rb_nvmem.c

```diff
@@ -0,0 +1,214 @@
+// SPDX-License-Identifier: GPL-2.0-only
+
+#include <linux/etherdevice.h>
+#include <linux/mod_devicetable.h>
+#include <linux/mtd/mtd.h>
+#include <linux/nvmem-consumer.h>
+#include <linux/nvmem-provider.h>
+#include <linux/platform_device.h>
+
+#define RB_MAGIC_HARD		"Hard"
+#define RB_HARDCONFIG_LEN 	0x1000
```

Comment:
This pending patch exports nvmem device size instead:
https://lore.kernel.org/lkml/20230718160652.26756-2-zajec5@gmail.com/

Reply:
Well, the thing is that patch has not moved for almost a year and I dont want to carry it though it was my idea as well to fetch the info from the MTD device, however there is no way to get the underlying MTD device from a NVMEM layout device as they intentionally made the structure internal to force people to go through helpers.

Reply:
hard_config is 0x2000 on e.g. ipq4018 devices: this code will not work everywhere. Can't you get the size from DTB at a minimum? default to 0x1000 if not set?

Reply:
We need to get the underlying MTD device here as only it has the size, there was no way to get its node as they walled of a the underlying NVMEM structure, but we can get the layout node and then get its parent (the partition node) and this works for getting the MTD size to make this dynamic.

Reply:
I had not noticed `nvmem_dev_size` went into 6.8. and has been backported (thanks Rafał) into OpenWrt for some time: https://github.com/openwrt/openwrt/blob/2db6459ed4e27939e1b72472eb22a9b979599d39/target/linux/generic/backport-6.6/819-v6.8-0010-nvmem-core-add-nvmem_dev_size-helper.patch#L29 https://github.com/openwrt/openwrt/commit/04ed95d1f92058ec26826791e309ec3ae09ee625

so could use that to avoid doing the mtd size workarounds.